pkg/detectors/sourcegraph/sourcegraph.go

```diff
@@ -0,0 +1,94 @@
+package sourcegraph
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"sourcegraph", "sg_token"}) + `\b(sgp_[a-z0-9]{40})\b`)
```

Comment:
question: Do you know if tokens allow for characters from g-z? I've only ever seen sg tokens that are `[a-f0-9]` in which case we could tighten the regex a bit more. Otherwise, this works just as well.

Reply:
I'll verify that 👍🏻

Reply:
From my testing it does not appear to use `g-z`.